RIOT-OS/RIOT
# gnrc_sixlowpan_frag_rb: migrate more functions from gnrc_sixlowpan_frag

### Contribution description
While implementing selective fragment recovery I ran into the problem of the reassembly buffer being bound to tight to classic reassembly, which makes it basically impossible to use without it. As a follow up to #12318 I now move also the reassembly buffer functions private to `gnrc_sixlowpan_frag` to `gnrc_sixlowpan_frag_rb` and unified the naming scheme (this is why I marked it as an API change, though just names are changed). For the most part, this PR just moves around stuff and renames it and other things.



### Testing procedure
Everything should still compile.

`tests/gnrc_sixlowpan_frag` and `tests/unittests` should still compile and run (for the latter testing `tests-gnrc_sixlowpan_frag_vrb` is enough).

`make doc` still works.

`examples/gnrc_networking` should still work on a 6LoWPAN capable platform with packets >120 bytes (I did not verify this yet myself)j



### Issues/PRs references
Follow-up to #12318.
